test/e2e/framework/network/utils.go

```diff
@@ -1134,19 +1134,39 @@ func TestUnderTemporaryNetworkFailure(ctx context.Context, c clientset.Interface
 //	}
 func BlockNetwork(ctx context.Context, from string, to string) {
 	framework.Logf("block network traffic from %s to %s", from, to)
-	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
-	dropCmd := fmt.Sprintf("sudo iptables --insert %s", iptablesRule)
-	if result, err := e2essh.SSH(ctx, dropCmd, from, framework.TestContext.Provider); result.Code != 0 || err != nil {
-		e2essh.LogResult(result)
-		framework.Failf("Unexpected error: %v", err)
+	nftablesRule := fmt.Sprintf("OUTPUT ip daddr %s reject", to)
+	dropCmdNftablesRule := fmt.Sprintf("sudo nft add rule ip nat %s", nftablesRule)
+	if result, err := e2essh.SSH(ctx, dropCmdNftablesRule, from, framework.TestContext.Provider); result.Code != 0 || err != nil {
+		framework.Logf("Nft table rule failed. Blocking traffic using iptables rule")
+
+		iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
+		dropCmdIptablesRule := fmt.Sprintf("sudo iptables --insert %s", iptablesRule)
+
+		if result, err := e2essh.SSH(ctx, dropCmdIptablesRule, from, framework.TestContext.Provider); result.Code != 0 || err != nil {
+			e2essh.LogResult(result)
+			framework.Failf("Unexpected error: %v", err)
+		}
 	}
 }
 
 // UnblockNetwork unblocks network between the given from value and the given to value.
 func UnblockNetwork(ctx context.Context, from string, to string) {
 	framework.Logf("Unblock network traffic from %s to %s", from, to)
-	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
-	undropCmd := fmt.Sprintf("sudo iptables --delete %s", iptablesRule)
+
+	var undropCmd string
+
+	fetchHandleCmd := fmt.Sprintf("sudo nft -a list ruleset | grep 'ip daddr %s reject' | awk -F'# handle' '{print $2}'", to)
```

Comment:
The simpler thing would be to just delete the whole table (which can be done by name rather than needing to use a handle.)

Any test using this function would have to be marked `[Serial]` / `[Disruptive]` anyway, so you don't have to worry that there might be another parallel caller to `BlockNetwork`.